fusillade/utils/authorize.py

```diff
@@ -92,44 +102,68 @@ def format_resources(resources: List[str], resource_param: List[str], kwargs: Di
     return [resource.format_map(_rp) for resource in resources]
 
 
-def format_context_entries(context_entries, kwargs):
+context_type_mapping = {
+    str: 'string',
+    # int: 'numeric',  # TODO: add support
+    # float: 'numeric',  # TODO: add support
+    # bool: 'bool'  # TODO: add support
+}
+
+
+def format_context_entries(context_entries: Dict[str, str],
+                           kwargs: Dict[str, Union[List[str], str]],
+                           restricted: List[str] = None) -> List[Dict]:
     """
-    >>> context_entries={"fus:context": "user_name"}
-    >>> kwargs={'user_name': "bob"}
+    >>> context_entries={"fus:context": "user_name", "fus:groups": "group"}
+    >>> kwargs={'user_name': "bob", 'group': ['g1', 'g2']}
     >>> x = format_resources(context_entries, kwargs)
-    >>> x == {"fus:context": "bob"}
+    >>> x == [{"ContextKeyName": "fus:context","ContextKeyValues":["bob"],"ContextKeyType": "string"},
+    >>>       {"ContextKeyName": "fus:groups","ContextKeyValues":["g1", "g2"],"ContextKeyType": "stringList"}]
 
     :param context_entries:
     :param kwargs:
+    :param restricted: A list of context variables not allowed.
     :return:
     """
-    _ce = dict()
+    _ce = []
+    restricted = restricted if restricted else []
     for key, value in context_entries.items():
-        if isinstance(value, list):
-            temp = []
-            for i in value:
-                v = kwargs.get(i)
-                if isinstance(v, str):
-                    temp.append(v)
-            _ce[key] = temp
+        if key in restricted:
+            continue
         else:
             v = kwargs.get(value)
-            if isinstance(v, str):
-                _ce[key] = v
+            if v is None:
+                continue
+            elif isinstance(v, list):
+                v_type = type(v[0])
+                if all(isinstance(i, v_type) for i in v):
+                    _ce.append({
+                        'ContextKeyName': key,
+                        'ContextKeyValues': v,
+                        'ContextKeyType': f"{context_type_mapping[v_type]}List"
+                    })
+            else:
```

Comment:
Instead of an `else:`, consider explicitly putting something like `elif isinstance(v, tuple):` since it will error on some types.  It's also a bit confusing to me what it's parsing here.

Ditto the above comment for any empty var (`str`, `tuple`, etc.) that is not `None`.

Reply:
added KeyError try block to restricted_context_entries